src/Workspaces/CoreTest/SolutionTests/SolutionWithSourceGeneratorTests.cs

```diff
@@ -930,6 +930,18 @@ public async Task TestChangingGeneratorChangesChecksum(TestHost testHost)
 
 #if NET
 
+    private sealed class DoNotLoadAssemblyLoader : IAnalyzerAssemblyLoader
+    {
+        public static readonly IAnalyzerAssemblyLoader Instance = new DoNotLoadAssemblyLoader();
+
+        public void AddDependencyLocation(string fullPath)
+        {
+        }
+
+        public Assembly LoadFromPath(string fullPath)
+            => throw new InvalidOperationException();
```

Comment:
Consider putting a "This test should not be loading analyzer assemblies" so if somebody doesn't know what this is it'll be a bit clearer.